zebra-network/src/protocol/external/types.rs

```diff
@@ -42,9 +42,43 @@ impl From<Network> for Magic {
 pub struct Version(pub u32);
 
 impl Version {
-    /// Returns the minimum network protocol version for `network` and
+    /// Returns the minimum remote node network protocol version for `network` and
+    /// `height`. Zebra disconnects from peers with lower versions.
+    ///
+    /// # Panics
+    ///
+    /// If we are incompatible with our own minimum remote protocol version.
+    pub fn min_remote_for_height(network: Network, height: block::Height) -> Version {
+        let min_spec = Version::min_specified_for_height(network, height);
+
+        // shut down if our own version is too old
+        assert!(
+            constants::CURRENT_NETWORK_PROTOCOL_VERSION >= min_spec,
+            "Zebra does not implement the minimum specified {:?} protocol version for {:?} at {:?}",
+            NetworkUpgrade::current(network, height),
+            network,
+            height,
+        );
+
+        max(min_spec, Version::initial_min_for_network(network))
+    }
+
+    /// Returns the minimum supported network protocol version for `network`.
+    /// This is the minimum peer version during initial block download.
```

Comment:
Nit: Not sure if this is what you intended, but maybe separate the sentences so that the first one is the summary and the second is in the expanded documentation?

```suggestion
    /// Returns the minimum supported network protocol version for `network`.
    ///
    /// This is the minimum peer version during initial block download.
```